### PR TITLE
Noinsync followers

### PR DESCRIPTION
**Pull request by mchacki**
Added debug script to identify situation of non-in-sync followers.

### CODE REVIEW

File: cleanup/lib/force-failover.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+  const res = arango.POST('/_api/agency/write', JSON.stringify([[data, prec]]));
+  if (res.results[0] === 0) {
+    print("WARNING: pre-condition failed, maybe cleanup already done");
+  } else {
+    print("INFO: " + JSON.stringify(res));
+  }
+  
+
+/*
```
Comment (joerg84) [inline cleanup/lib/force-failover.js]: Delete comments

File: cleanup/force-failover.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+set -u
+
+# usage: remove-zombie.sj AGENCY_ENDPOINT ZOMBIE_FILE
```
Comment (joerg84) [inline cleanup/force-failover.sh]: The usage seems for another script

File: cleanup/force-failover.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+# The sript requires an installed version of arangodb or at least a way to call
```
Comment (joerg84) [inline cleanup/force-failover.sh]: s/sript/script

**Comment by sleto-it** [comment]
Hi,

What is the status of this PR - is this check still current and useful?

If yes, as I suspect, I'd suggest we resolve the conflict and merge
What is blocking? e.g. review?

Thanks,

**Comment by sleto-it** [comment]
hi all - sending a new message - can this be reviewed, conflict resolved and merged - or closed if outdated?

thanks

File: cleanup/lib/force-failover.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    print("usage: force-failover.sh --server.endpoint LEADER-AGENT INPUT-FILE TARGET-SERVER LEADER-CID SHARD-INDEX");
+    return;
+  }
+
+  try {
+    let role = require("@arangodb").db._version(true).details.role;
+
+    if (role === "AGENT") {
+      let agency = arango.POST('/_api/agency/read', [
+        ["/"]
```
Comment (maierlars) [inline cleanup/lib/force-failover.js]: Do we really want to read the **WHOLE** agency to test if this is the leader? :D you could also use '/_api/agency/config' and compare `leaderId` with `configuration.id`.

File: cleanup/lib/force-failover.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      print(`FATAL: Given shardindex out of bounds, given index: ${shardIndex} numberOfShards found: ${colInfo.length}`);
+      return;
+    }
+    const myinfo = colInfo[shardIndex];
+    const planPathPrefix = `/arango/Plan/Collections/${db}/${cid}/shards/${myinfo.shard}`;
+    const oldServers = myinfo.servers;
+    const desiredServers = myinfo.servers
+      .slice(1) // remove old leader
+      .filter(s => s !== target); // remove the desired leader from the position it is in
+    desiredServers.unshift(target); // add desired leader in front (inplace operation)
```
Comment (maierlars) [inline cleanup/lib/force-failover.js]: Did I miss the check that the new leader is actually in sync? And in Plan?

File: cleanup/lib/force-failover.js (hunk 4 above)
Comment (mchacki) [reply cleanup/lib/force-failover.js]: oh this is a good point !
I think it is enough that this agent actually responds, so we could read `/foo` instead which is comparably cheap.
However this is not good in all scripts, so I would like to extract this into a different PR

File: cleanup/lib/force-failover.js (hunk 5 above)
Comment (mchacki) [reply cleanup/lib/force-failover.js]: This information has been generated by the analysis script.
in the oldServers there is all information about the plan.
The purpose of this force failover is that we can failover to a server that is NOT in sync. (the target)

**Pull request merged**